core/service/src/chain_ops.rs

```diff
@@ -103,13 +118,18 @@ pub fn import_blocks<F, E, R>(
 ) -> error::Result<()>
 	where F: ServiceFactory, E: Future<Item=(),Error=()> + Send + 'static, R: Read,
 {
-	struct DummyLink;
-	impl<B: Block> Link<B> for DummyLink { }
-
 	let client = new_client::<F>(&config)?;
 	// FIXME #1134 this shouldn't need a mutable config.
 	let queue = components::FullComponents::<F>::build_import_queue(&mut config, client.clone())?;
-	queue.start(DummyLink)?;
+
+	let (wait_send, wait_recv) = std::sync::mpsc::channel();
+	impl<B: Block> Link<B> for WaitLink {
+		fn block_imported(&self, _hash: &B::Hash, _number: NumberFor<B>) {
+			self.wait_send.send(()).expect("Unable to notify about imported block");
```

Comment:
Can you provide a proof why this expect won't fail?

Reply:
Hmm hmm, this would panic if the receiving side in the main thread is no longer there ‒ but then the process, incl. the thread where this `send` is invoked, would be dead either way. Does this sound like an acceptable proof to you?

So basically it can't happen because when the receiver in the main thread is no longer there then this thread would also be dead.